src/_macosx.m

```diff
@@ -394,9 +394,20 @@ static CGFloat _get_device_scale(CGContextRef cr)
 static PyObject*
 FigureCanvas_flush_events(FigureCanvas* self)
 {
-    // We need to allow the runloop to run very briefly
-    // to allow the view to be displayed when used in a fast updating animation
-    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.0]];
+    // We run the app, matching any events that are waiting in the queue
+    // to process, breaking out of the loop when no events remain and
+    // displaying the canvas if needed.
+    NSEvent *event;
+    while (true) {
+        event = [NSApp nextEventMatchingMask: NSEventMaskAny
+                                   untilDate: [NSDate distantPast]
```

Comment:
I do not fully understand https://developer.apple.com/documentation/appkit/nsapplication/1428485-nexteventmatchingmask . It is not clear if "expiration" here means "do not wait longer than this" or "do not return any events newer than this".  If it is the second, I think grabbing the current time outside of the `while` loop and passing it here will prevent the perpetual-motion machine I am worried about.

Reply:
See recent commit. The `distantFuture` discussion I think indicates the second of those. https://developer.apple.com/documentation/foundation/nsdate/1415385-distantfuture

Reply:
> You can use the object returned by distantFuture as the date argument to wait indefinitely for the event to occur.

I read this as pointing to the first (it is a time out not a filter)?